url4c41254592_topskychen_AuthTopk_tgz-pJ8-quickhull3d_QuickHull3DTestJ8/src/spatialindex/rtree/RTree.java

```diff
@@ -0,0 +1,1674 @@
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+//		System.out.println("id :\t" + id);
+		DataOfPoint dataOfPoint = new DataOfPoint(data);
+		if(VO.isSoloAuth){
+			buffer_Points.put(id, dataOfPoint);
+		}
+		return dataOfPoint;
+	}
+	
+	public DataOfLine loadDataOfLineFromIndex(long id1, long id2){
+		long id = DataOfLine.calcLineId(id1, id2);
+		if(VO.isSoloAuth){
+			if(buffer_Lines.containsKey(id))return buffer_Lines.get(id);
+		}
+		if(tOfLine == null){
+			loadIndexOfLines();
+		}
+		if(tOfLine.containsKey(id) == false){
+			//System.err.println("Not found line:\t" + id1 + ", " + id2);
+//			System.out.print("x");
+			return null;
+		}
+		long[] tmp = tOfLine.get(id);
+		long pos = tmp[0], len = tmp[1];
+		byte[] data = new byte[(int)len];
+		try {
+			rafLine.seek(pos);
+			rafLine.read(data);
+		} catch (IOException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		DataOfLine dataOfLine = new DataOfLine(data);
+		if(VO.isSoloAuth){
+			buffer_Lines.put(id, dataOfLine);
+		}
+		return dataOfLine;
+	}
+	
+	public DataOfLine loadDataOfLineFromBtree(int id1, int id2){
+		long id = DataOfLine.calcLineId(id1, id2);
+		if(VO.isSoloAuth){
+			if(buffer_Lines.containsKey(id))return buffer_Lines.get(id);
+		}
+		if(tOfLine == null){
+			loadIndexOfLines();
+		}
+		if(tOfLine.containsKey(id) == false){
+//			System.out.print("x");
+//			System.err.println("Not found line:\t" + id1 + ", " + id2);
+			return null;
+		}
+		long[] tmp = tOfLine.get(id);
+		long pos = tmp[0], len = tmp[1];
+		byte[] data = new byte[(int)len];
+		try {
+			rafLine.seek(pos);
+			rafLine.read(data);
+		} catch (IOException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		DataOfLine dataOfLine = new DataOfLine(data);
+		if(VO.isSoloAuth){
+			buffer_Lines.put(id, dataOfLine);
+		}
+		return dataOfLine;
+	}
+	
+	/**
+	 * ps contains FillFactor, IndexCapacity, LeafCapacity, Dimension. Sm is the
+	 * io manager of r-tree file
+	 * 
+	 * @param ps
+	 * @param sm
+	 */
+	public RTree(PropertySet ps, IStorageManager sm) {
+		m_rwLock = new RWLock();
+		m_pStorageManager = sm;
+		m_rootID = IStorageManager.NewPage;
+		m_headerID = IStorageManager.NewPage;
+		m_treeVariant = SpatialIndex.RtreeVariantRstar;
+		m_fillFactor = 0.7f;
+		m_indexCapacity = 100;
+		m_leafCapacity = 100;
+		m_nearMinimumOverlapFactor = 32;
+		m_splitDistributionFactor = 0.4f;
+		m_reinsertFactor = 0.3f;
+		m_dimension = 2;
+
+		m_infiniteRegion = new Region();
+		m_stats = new Statistics();
+
+		destFileNamePoint = (String)ps.getProperty("destFileNamePoint");
+		destFileNameLine = (String)ps.getProperty("destFileNameLine");
+		
+		Object var = ps.getProperty("IndexIdentifier");
+		if (var != null) {
+			if (!(var instanceof Integer)) {
+				throw new IllegalArgumentException("Property IndexIdentifier must an Integer");
+			}
+			m_headerID = ((Integer) var).intValue();
+			try {
+				initOld(ps);
+			} catch (IOException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 536.